CHANGELOG.md

```diff
@@ -53,6 +53,7 @@
 * [9495](https://github.com/grafana/loki/pull/9495) **thampiotr**: Promtail: Fix potential goroutine leak in file tailer.
 * [9650](https://github.com/grafana/loki/pull/9650) **ashwanthgoli**: Config: ensure storage config defaults apply to named stores.
 * [9629](https://github.com/grafana/loki/pull/9629) **periklis**: Fix duplicate label values from ingester streams.
+* [9754](https://github.com/grafana/loki/pull/9754) **ashwanthgoli**: store: fix incorrect usage of period config in table range
```

Comment:
nit: Please phrase the changelog in a way that the user can identify it as a fix to their problem.

```
Store: Fix an issue where different index prefixes in the schemas period configs leads to missing logs from the previous schema periods when querying logs that have been ingested during a previous schema period.

Reply:
It could be condensed a little bit, though.

Reply:
how does this sound? ```Fixes an issue where indexes using a prefix different from the one in current period become unqueriable```